packages/report-server/src/reportBuilder/functions/math.ts

```diff
@@ -1,62 +1,20 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
 /**
  * Tupaia
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { sum as mathjsSum, divide as mathjsDivide } from 'mathjs';
-
-const isNumeric = (value: unknown): value is number | undefined =>
-  value === undefined || typeof value === 'number';
-const isNumericOrArray = (
-  value: unknown,
-): value is number | undefined | Array<number | undefined> =>
-  isNumeric(value) || (Array.isArray(value) && value.every(isNumeric));
-const isDefined = <T>(value: T): value is Exclude<T, undefined> => value !== undefined;
-
-export const sum = (
-  ...args: (number | undefined | (number | undefined)[])[]
-): number | undefined => {
-  if (!args.every(isNumericOrArray)) {
-    throw new Error('sum received invalid input type');
-  }
-
-  const numbers = args.flat();
-
-  const validNumbers = numbers.filter(isDefined);
-
-  if (validNumbers.length === 0) {
-    return undefined;
-  }
-
-  if (validNumbers.length === 1) {
-    return validNumbers[0];
-  }
-
-  return mathjsSum(validNumbers);
-};
-
-export const divide = (
-  ...args: (number | undefined | (number | undefined)[])[]
-): number | undefined => {
-  if (!args.every(isNumericOrArray)) {
-    throw new Error('divide received invalid input type');
-  }
-
-  const numbers = args.flat();
-
-  if (numbers.some(number => !isDefined(number))) {
-    return undefined;
-  }
-
-  const validNumbers = numbers.filter(isDefined);
-
-  if (validNumbers.length === 0) {
-    return undefined;
-  }
+import { typed } from 'mathjs';
 
-  if (validNumbers.length === 1) {
-    return validNumbers[0];
-  }
+export const divide = typed('divide', {
+  'number, undefined': (num: number, undef: undefined) => undefined,
+  'undefined, number': (undef: undefined, num: number) => undefined,
+  'undefined, undefined': (undef: undefined, undef2: undefined) => undefined,
+});
 
-  return validNumbers.reduce((numerator, denominator) => mathjsDivide(numerator, denominator));
-};
+export const add = typed('add', {
+  'number, undefined': (num: number, undef: undefined) => num,
+  'undefined, number': (undef: undefined, num: number) => num,
+  'undefined, undefined': (undef: undefined, undef2: undefined) => undefined,
+});
```

Comment:
Okay bad news bears, seems that using this can't force the `mathjs.sum` function to return `undefined` if passed values that are all `undefined` (it returns `0` which isn't our expected result) 😢 So may still need to manually override `sum` the old fashioned way... But still will be very happy to be able to use `$a + $b` without worry if they're defined or not 🎉

Reply:
Bugger!